programs/fileio.h

```diff
@@ -96,22 +103,23 @@ void FIO_setNotificationLevel(int level);
 void FIO_setExcludeCompressedFile(FIO_prefs_t* const prefs, int excludeCompressedFiles);
 void FIO_setPatchFromMode(FIO_prefs_t* const prefs, int value);
 void FIO_setContentSize(FIO_prefs_t* const prefs, int value);
-void FIO_setNbFiles(FIO_prefs_t* const prefs, int value);
-void FIO_setCurrFileIdx(FIO_prefs_t* const prefs, int value);
+
+/* FIO_ctx_t functions */
+void FIO_setNbFilesTotal(FIO_ctx_t* const fCtx, int value);
 
 /*-*************************************
 *  Single File functions
 ***************************************/
 /** FIO_compressFilename() :
  * @return : 0 == ok;  1 == pb with src file. */
-int FIO_compressFilename (FIO_prefs_t* const prefs,
+int FIO_compressFilename (FIO_prefs_t* const prefs, FIO_ctx_t* const fCtx,
```

Comment:
The comment regarding parameter order is a little bit more important here
because it's applied to public functions, exposed via `fileio.h`.

Also, a question : is `prefs` still mutable ?

Reply:
After this PR, none of my recent changes should mutate `prefs`, but I'm not sure if there's somewhere else in the code that might modify `prefs` during (de)compression.

In a followup refactor PR focused more on making full use of `fCtx` I will also try to modify the relevant function signatures to use `const FIO_prefs_t* const prefs` to ensure it's not mutable.